{{cookiecutter.ioc_name}}/activate_env.sh

```diff
@@ -1,6 +1,34 @@
 #!/bin/bash
 
+if [ -f "/reg/g/pcds/pyps/conda/pcds_conda" ]; then
+  echo "* Sourcing pcds_conda to get access to conda"
+  # shellcheck disable=SC1091
+  source /reg/g/pcds/pyps/conda/pcds_conda
+fi
+
+if [ ! -f "config.sh" ]; then
+  echo "* IOC configuration not found!"
+  exit 1
+fi
+
+echo "* Sourcing this IOC's configuration"
 . config.sh
 
-conda deactivate
-conda activate "$CONDA_ENV_PATH"
+echo "  - IOC:                     $IOC"
+echo "  - IOC_COMMAND:             $IOC_COMMAND"
+echo "  - PREFIX:                  $PREFIX"
+echo "  - IOC_DATA_PATH:           $IOC_DATA_PATH"
+echo "  - CONDA_ENV_PATH:          $CONDA_ENV_PATH"
+echo "  - EPICS_CA_AUTO_ADDR_LIST: $EPICS_CA_AUTO_ADDR_LIST"
+echo "  - EPICS_CA_ADDR_LIST:      $EPICS_CA_ADDR_LIST"
+echo "  - CAPTURE_LOGS:            $CAPTURE_LOGS"
+echo "  - LOG_FILE_PATH:           $LOG_FILE_PATH"
+
+echo "* Deactivating any existing conda environment"
+conda deactivate || echo "Conda deactivate failed; continuing..."
```

Comment:
Apparently if you `conda deactivate` on the base environment, you might lose access to conda entirely.  [The guidance ](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#deactivating-an-environment)is `conda activate` with no additional argument.  (TIL)

Also what was the case where `conda deactivate` was failing?  Was pcds-conda not getting sourced correctly?

Reply:
Ah, that's new to me (or new to me _again_ due to faulty memory) - thanks for the tip! I'll change that.

This may have been just a temporary band-aid for the issue that Will had encountered. It may not have actually been failing. I hope with the suggestion it won't be an issue going forward.